### PR TITLE
CI: ax.rowNum and ax.colNum attributes deprecated in Matplotlib 3.2

### PR DESCRIPTION
**Pull request by simonjayhawkins**
https://github.com/pandas-dev/pandas/pull/32442/checks?check_run_id=485569958

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
@WillAyd not sure about this. should we just be silencing the deprecation warnings instead. `The rowNum attribute was deprecated in Matplotlib 3.2 and will be removed two minor releases later. Use ax.get_subplotspec().rowspan.start instead.`

**Comment by WillAyd** [comment]
On a phone so haven’t looked deeply but does swapping out the attribute access for what the message says not work?

**Comment by simonjayhawkins** [comment]
> On a phone so haven’t looked deeply but does swapping out the attribute access for what the message says not work?

it doesn't work on older versions. hopefully latest commit fixes but is more complex (may not be relevant).

The alternative is silence the warnings and fix at a later date when the attributes have been deprecated.

**Comment by simonjayhawkins** [comment]
> The alternative is silence the warnings and fix at a later date when the attributes have been deprecated.

assuming our minimum supported version has 'SubplotSpec' object has attribute 'rowspan' by then.

**Comment by simonjayhawkins** [comment]
hmm, appears new matplotlib related errors on Linux py37_locale.

**Comment by simonjayhawkins** [comment]
right. fixed warnings in web and docs, but for some reason have two new test failures. have skipped so we can get to ci to green, raise an issue, and resolve independently

**Comment by WillAyd** [comment]
@simonjayhawkins how about just capping matplotlib for now to get CI green then can come back to this at a deeper level (?)

File: pandas/plotting/_matplotlib/tools.py [hunk 1]
```diff
@@ -288,20 +290,26 @@ def _remove_labels_from_axis(axis):
 
 def _handle_shared_axes(axarr, nplots, naxes, nrows, ncols, sharex, sharey):
     if nplots > 1:
+        if compat._mpl_ge_3_2_0():
+            row_num = lambda x: x.get_subplotspec().rowspan.start
+            col_num = lambda x: x.get_subplotspec().rowspan.start
```
Comment (datapythonista) [inline pandas/plotting/_matplotlib/tools.py]:
```suggestion
            col_num = lambda x: x.get_subplotspec().colspan.start
```

**Review by datapythonista — COMMENTED** [review]
There is a typo.

And probably worth adding a test, since this only failed in the docs. The next code is what the docs have, calls the affected code and should generate the warning. I think it should be a test good enough (or if you prefer a unit test that calls `_handle_shared_axes` that should be ok too):

```python
import numpy as np
import pandas as pd
from matplotlib import pyplot as plt

ts = pd.Series(np.random.randn(1000), index=pd.date_range('1/1/2000', periods=1000))
ts = ts.cumsum()

df = pd.DataFrame(np.random.randn(1000, 4), index=ts.index, columns=list('ABCD'))
df = df.cumsum()

fig, axes = plt.subplots(4, 4, figsize=(6, 6))
plt.subplots_adjust(wspace=0.5, hspace=0.5)
target1 = [axes[0][0], axes[1][1], axes[2][2], axes[3][3]]
target2 = [axes[3][0], axes[2][1], axes[1][2], axes[0][3]]

df.plot(subplots=True, ax=target1, legend=False, sharex=False, sharey=False);
```

**Comment by simonjayhawkins** [comment]
> There is a typo.

nice catch. Thanks @datapythonista

**Comment by simonjayhawkins** [comment]
> And probably worth adding a test

ok as a follow-up if this is green?

**Comment by datapythonista** [comment]
> ok as a follow-up if this is green?

Sure, the docs are being our tests already. Just thought that would be good to see this warning failing the tests. But happy to see the CI fixed asap.

**Comment by simonjayhawkins** [comment]
> But happy to see the CI fixed asap.

yeah, if this fails, will go with @WillAyd suggestion, https://github.com/pandas-dev/pandas/pull/32444#issuecomment-594967595

**Pull request merged**